### ISSUE TITLE
URL encoding in search result images' `src`

### ISSUE THREAD

**Issue by n-older** [CloudCannon/pagefind#743]
We use Netlify's Image CDN to transform images but when we set the `image` key for `data-pagefind-meta` to the transformed image URL using query parameters, Pagefind seems to be encoding the `&`s into `&amp;` when rendering the image in the search results, thus breaking the URL/image.

For example:

```https://emailable.com/.netlify/images?fit=cover&fm=avif&h=720&url=https%3A%2F%2Fsanity-cdn.emailable.com%2Fproduction%2F0e09167ca2c53bc385def9f838d6c3f7d41a59a6-1600x900.png&w=1280```

gets turned into

```https://emailable.com/.netlify/images?fit=cover&amp;fm=avif&amp;h=720&amp;url=https%3A%2F%2Fsanity-cdn.emailable.com%2Fproduction%2F0e09167ca2c53bc385def9f838d6c3f7d41a59a6-1600x900.png&amp;w=1280```

**Comment by n-older** [comment]
Hi @bglw

I unfortunately can't provide a link but I can provide snippets of the source code and the rendered HTML if that helps.

Source code for generating the page's image (some non-related arguments omitted for brevity):
```erb
<%= image_tag(
  data.featured_image,
  alt: data.title,
  data: {
    'pagefind-meta': "image:#{image_path(data.featured_image, h: 90, w: 160)}"
  }
) %>

Rendered HTML for page's image with metadata:
```html
<img class="mt-[3.75rem] md:mt-20 mx-auto rounded-lg w-full md:max-w-5xl md:rounded-3xl" alt="Ultimate Email Deliverability Guide for Black Friday 2024" fetchpriority="high" data-pagefind-meta="image:/.netlify/images?fm=avif&amp;h=90&amp;url=https%3A%2F%2Fsanity-cdn.emailable.com%2Fproduction%2F0e09167ca2c53bc385def9f838d6c3f7d41a59a6-1600x900.png&amp;w=160,image_alt:Ultimate Email Deliverability Guide for Black Friday 2024" src="/.netlify/images?fit=cover&amp;fm=avif&amp;h=1800&amp;url=https%3A%2F%2Fsanity-cdn.emailable.com%2Fproduction%2F0e09167ca2c53bc385def9f838d6c3f7d41a59a6-1600x900.png&amp;w=3200" width="1600" height="900" sizes="(max-width: 900px) calc(100vw - 40px), 840px" srcset="/.netlify/images?fit=cover&amp;fm=avif&amp;h=180&amp;url=https%3A%2F%2Fsanity-cdn.emailable.com%2Fproduction%2F0e09167ca2c53bc385def9f838d6c3f7d41a59a6-1600x900.png&amp;w=320 320w, /.netlify/images?fit=cover&amp;fm=avif&amp;h=360&amp;url=https%3A%2F%2Fsanity-cdn.emailable.com%2Fproduction%2F0e09167ca2c53bc385def9f838d6c3f7d41a59a6-1600x900.png&amp;w=640 640w, /.netlify/images?fit=cover&amp;fm=avif&amp;h=540&amp;url=https%3A%2F%2Fsanity-cdn.emailable.com%2Fproduction%2F0e09167ca2c53bc385def9f838d6c3f7d41a59a6-1600x900.png&amp;w=960 960w, /.netlify/images?fit=cover&amp;fm=avif&amp;h=720&amp;url=https%3A%2F%2Fsanity-cdn.emailable.com%2Fproduction%2F0e09167ca2c53bc385def9f838d6c3f7d41a59a6-1600x900.png&amp;w=1280 1280w, /.netlify/images?fit=cover&amp;fm=avif&amp;h=900&amp;url=https%3A%2F%2Fsanity-cdn.emailable.com%2Fproduction%2F0e09167ca2c53bc385def9f838d6c3f7d41a59a6-1600x900.png&amp;w=1600 1600w, /.netlify/images?fit=cover&amp;fm=avif&amp;h=1080&amp;url=https%3A%2F%2Fsanity-cdn.emailable.com%2Fproduction%2F0e09167ca2c53bc385def9f838d6c3f7d41a59a6-1600x900.png&amp;w=1920 1920w">
```

Rendered HTML for search result image:
```html
<img class="pagefind-ui__result-image svelte-4xnkmf" src="/.netlify/images?fm=avif&amp;amp;h=90&amp;amp;url=https%3A%2F%2Fsanity-cdn.emailable.com%2Fproduction%2F0e09167ca2c53bc385def9f838d6c3f7d41a59a6-1600x900.png&amp;amp;w=160,image_alt:Ultimate Email Deliverability Guide for Black Friday 2024" alt="Ultimate Email Deliverability Guide for Black Friday 2024">
```

**Comment by bglw** [comment]
Hi @n-older — thanks again for the report! Sorry for the delay in getting back to you. 

A fix for this has landed in [Pagefind v1.3.0](https://github.com/CloudCannon/pagefind/releases/tag/v1.3.0) 🎉 Let me know if you see any further issues 🙂
